src/core/server/server.api.md

```diff
@@ -31,7 +32,13 @@ import { ClusterPutSettingsParams } from 'elasticsearch';
 import { ClusterRerouteParams } from 'elasticsearch';
 import { ClusterStateParams } from 'elasticsearch';
 import { ClusterStatsParams } from 'elasticsearch';
+import { ConfigDeprecation } from '@kbn/config';
+import { ConfigDeprecationFactory } from '@kbn/config';
+import { ConfigDeprecationLogger } from '@kbn/config';
+import { ConfigDeprecationProvider } from '@kbn/config';
```

Comment:
Another issue moving these to a package: The doc files are no longer generated, as they are imported/exported from the package. See https://github.com/elastic/kibana/pull/76874/commits/545c867c7df793bc7ffc081abf22aeb69f0d4f2d. Not sure if we have a solution for that

Reply:
I don't think we do. `@kbn/config` becomes an external package for `src/core`, so `api-extractor` terminates traversing at this point. ~Investigation required to understand whether we can link types between packages.~ It doesn't  supported https://github.com/microsoft/rushstack/issues/1596

Reply:
So by extracting `config`, and `logging` outside of core, we basically are loosing the documentation for their types. 

Would generating doc from these package solve this? I guess `core` and `@kbn/xxx` generated doc would not be able to link to each other...

Reply:
>  I guess core and @kbn/xxx generated doc would not be able to link to each other...

Right, in the issue I attached discussed that this use-case is not supported out-of-the-box. We might have to contribute to `api-extractor` to support this.

Reply:
Should we create an issue to contribute to `api-extractor` ? @joshdover

Reply:
Created https://github.com/elastic/kibana/issues/77529 as a discussion issue